src/Runners/TestSuiteLoader.ts

```diff
@@ -19,6 +19,25 @@ export default class TestSuiteLoader {
         return suites;
     }
 
+    async loadSelectedTestSuites(dir: string, filter: string): Promise<Record<string, TestSuite>> {
+        const files = (await this.file_system.getFiles(dir))
+            .filter((file) => FileSystem.extension(file) === FileSystem.extension(__filename));
+        const suites: Record<string, TestSuite> = {};
+        for (const file of files) {
+            const [, relativePathToSuite] = file.split(dir + path.sep);
+            const suiteName = FileSystem.fileName(relativePathToSuite);
```

Comment:
I think requiring that the filename match the class name might be too limiting

Reply:
Agreed. This was an attempt to prevent loading the test suite if it's not in the filter, but I hadn't considered that this adds some unintentional restrictions. I've reworked this in 43b6a8b to always load the test suite and _then_ filter by class name instead. Please let me know your thoughts.